### DISCUSSION TITLE
Dialog (Modal) - How to disable autofocus when opened?

### DISCUSSION THREAD

**Discussion by binhxn** [ariakit/ariakit#985] [Questions] [answered]
Hello 👋 ,

I was wondering if it's possible to disable autofocus (or hide the first focus ring outline) when a modal is visible? My goal is to only display focus ring on the first tabbable element **_after_** pressing tab when the modal is visible. Approaches I've tried:

1. I initially attempted to set `tabIndex={0}` on a non-tabbable element (e.g. the modal container), but as [mentioned in the docs](https://reakit.io/docs/dialog/#initial-focus), it was included in the tab order.
2. I tried to work my way around `unstable_initialFocusRef` but that didn't work either.

Any thoughts on how to go about this?

**Comment by diegohaz** [comment, marked as answer]
You should be able to use the [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) pseudo-class (or a [polyfill](https://github.com/WICG/focus-visible)).

**Reply by binhxn** [reply]
Hey @diegohaz ,

Thanks for the quick response. I realized I was asking two different questions and apologize for any confusion. Your answer helped me resolved the focus-ring issue, which is great 👍 . But how would I go about disabling auto focus in general? Here's a contrived example:

```jsx
<Modal>
  <button>Close</button>
  <input type="text" placeholder="First Name" />
</Modal>
```

Based on the above, when the modal is visible, nothing is focused yet. But once I press tab, the Close button will be focused first. The focus should only be tabbable between button and input as well. Any thoughts on how to create this behavior?

**Reply by diegohaz** [reply]
You should be able to focus on the container first and use `:focus-visible` to show the focus ring only when the user presses tab.
```jsx
const ref = useRef();
<Dialog ref={ref} unstable_initialFocusRef={ref}>
  ...
</Dialog>
```

**Reply by binhxn** [reply]
That worked! I realized I was focused (no pun intended) on using refs on my components rather than on Reakit components. Thank you for your help!
